toolchain/check/eval.cpp

```diff
@@ -184,19 +235,36 @@ auto TryEvalInst(Context& context, SemIR::InstId inst_id, SemIR::Inst inst)
       return context.constant_values().Get(
           inst.As<SemIR::Converted>().result_id);
 
+    // `not true` -> `false`, `not false` -> `true`.
+    // All other uses of unary `not` are non-constant.
     case SemIR::UnaryOperatorNot::Kind: {
       auto const_id = context.constant_values().Get(
           inst.As<SemIR::UnaryOperatorNot>().operand_id);
-      if (!const_id.is_template()) {
-        break;
+      auto phase = GetPhase(const_id);
+      if (phase == Phase::Template) {
+        auto value =
+            context.insts().GetAs<SemIR::BoolLiteral>(const_id.inst_id());
+        value.value =
+            (value.value == SemIR::BoolValue::False ? SemIR::BoolValue::True
+                                                    : SemIR::BoolValue::False);
+        return MakeConstantResult(context, value, Phase::Template);
+      }
+      if (phase == Phase::Error) {
+        return SemIR::ConstantId::Error;
+      }
+      break;
+    }
+
+    // `const (const T)` evaluates to `const T`. Otherwise, `const T` evaluates
+    // to itself.
```

Comment:
This use of `const` feels like it may be confusing if we use `const` for runtime constants.

Reply:
Resolving based on in-person discussion.